libraries/rush-lib/src/cli/RushXCommandLine.ts

```diff
@@ -44,129 +56,150 @@ interface IRushXCommandLineArguments {
   commandArgs: string[];
 }
 
+class ProcessError extends Error {
+  public exitCode: number = 0;
+  public constructor(message: string, exitCode: number) {
+    super(message);
+    this.exitCode = exitCode;
+  }
+}
+
 export class RushXCommandLine {
-  public static launchRushX(launcherVersion: string, isManaged: boolean): void {
-    RushXCommandLine._launchRushXInternal(launcherVersion, { isManaged });
+  public static launchRushX(launcherVersion: string, options: ILaunchRushXInternalOptions): void {
+    try {
+      const args: IRushXCommandLineArguments = this._getCommandLineArguments();
+      const rushConfiguration: RushConfiguration | undefined = RushConfiguration.tryLoadFromDefaultLocation({
+        showVerbose: false
+      });
+      const eventHooksManager: EventHooksManager | undefined = rushConfiguration
+        ? new EventHooksManager(rushConfiguration)
+        : undefined;
+
+      const suppressHooks: boolean = process.env._RUSH_SUPPRESS_HOOKS === '1';
```

Comment:
Why not make this a standard env variable and put it in https://github.com/microsoft/rushstack/blob/main/libraries/rush-lib/src/api/EnvironmentConfiguration.ts.